### ISSUE TITLE
RFE: build llvm / clang / lld / openmp with separate build scripts

### ISSUE THREAD

**Issue by tomty89** [termux/termux-packages#2282]
If you take a look at the file lists of the current libllvm subpackages, you will notice that the package split is a total chaos. It has been problematic during the 5.0-time and is now nothing but awful.

This is because the file names aren't always in straight forward patterns that makes it simple to "redivide" the packages, and because of how package split is done in Termux (that we can't have procedures to copy from different locations of the build but move files away from an already-merged source), it gets even nastier when files are accidentally left in the current single main package, libllvm.

If we build them with separate scripts, at worst the files will only be scattered around their own subpackage, which makes it way easier to fix (in terms of both identifying where the files should belong to and conflicts/replaces matter after the fix).

And we will have sane -dev package to main package dependency that way. (And probably sane auto -dev split)

Also it doesn't seem to me that having the "lib" package as the main package is a good idea for either llvm or clang, considering their nature and/or the hierarchy of their splits.

For clang, I am not even sure it should be split at all. Most likely we would at most want to split certain header files and static library (.a) to its dev packages (maybe it will be automatically done, idk), but not a silly "libclang" subpackage. (Though, as long as it's not the main package, it wouldn't be too ugly to have libclang.so split, but I really doubt that it has a point.)

I would love to fix this myself when I get regular access to a PC (currently an aarch64 android phone is all I have). It would be great if someone gets ahead of me though.

**Comment by xtkoba** [comment]
This should be possible. Gentoo Portage has separate build recipes for these components (not exhaustive):

```
dev-util/lldb
sys-devel/clang
sys-devel/lld
sys-devel/llvm
sys-libs/compiler-rt
sys-libs/compiler-rt-sanitizers
sys-libs/libcxx
sys-libs/libcxxabi
sys-libs/libomp
sys-libs/llvm-libunwind
```

Rebuilding everything just for only one component (e.g. `lld` after `python` bump) is nothing but ridiculous.

The only drawback that I think of is that it may be difficult to align the version among the components.

**Comment by licy183** [comment]
I agree with this. Separating packages will not only save lots of build time but also make it easier to maintain. 
`flang` has already been separated from `libllvm` and it build successfully, I suppose `clang` will too.

Ping: @buttaface

**Comment by buttaface** [comment]
It is possible, but as I just wrote in a comment on that closed mingw pull:
```
We would save build time building LLVM separately,
but it's more work setting all that up and maintaining it,
so I don't think it is worth it,
as build time is much cheaper than our time.
```
